pinot-segment-local/src/main/java/org/apache/pinot/segment/local/realtime/impl/json/MutableJsonIndexImpl.java

```diff
@@ -342,6 +323,51 @@ public String[] getValuesForKeyAndDocs(int[] docIds, Map<String, RoaringBitmap>
     return values;
   }
 
+  private Pair<String, RoaringBitmap> processArrayIndex(String key) {
```

Comment:
`processArrayIndex` -> `getKeyAndFlattenedDocIds`?

Can you add a comment here about input/output (already done in the immutable reader)